algorithms/unit_tests/TestSort.hpp

```diff
@@ -402,6 +403,330 @@ void test_sort_integer_overflow() {
       << "view (" << vh[0] << ", " << vh[1] << ") is not sorted";
 }
 
+// Comparator for sorting in descending order
+template <typename Key>
+struct GreaterThan {
+  KOKKOS_INLINE_FUNCTION bool operator()(const Key lhs, const Key rhs) const {
+    return lhs > rhs;
+  }
```

Comment:
```suggestion
  KOKKOS_FUNCTION static constexpr bool operator()(const &Key lhs, const &Key rhs) {
    return lhs > rhs;
  }
```

Reply:
@dalg24 I agree about these changes, except for ``static`` - the sorts take an instance of the comparison functor (for maximum flexibility - this lets the functor carry extra data like Views or whatever if needed). Also, the ``StdAlgoLessThanBinaryPredicate`` that is built in (and now used as the default comparison) has non-static operator().

Reply:
BTW, Thrust sort/sort_by_key that takes a ``StrictWeakOrdering`` is also an instance, and expects non-static